docs/integrations/iceberg.md

```diff
@@ -2,20 +2,343 @@
+spark.sql("SELECT * FROM my_table").show()
+
+// Switch to a feature branch
+spark.sql("USE my_repo.new_branch.namespace")
+spark.sql("SELECT * FROM my_table").show()
+```
+
+  </div>
+
+  <div markdown="4" id="starrocks">
+
+```sql
+CREATE EXTERNAL CATALOG lakefs
+COMMENT "lakeFS REST Catalog"
+PROPERTIES (
+    "type"                          = "iceberg",
+    "iceberg.catalog.type"          = "rest",
+    "iceberg.catalog.uri"           = "https://lakefs.example.com/catalog/iceberg/v1",
+    "client.factory"                = "com.starrocks.connector.iceberg.IcebergAwsClientFactory"
+);
+
+-- Use a specific repository and branch
+SELECT * FROM frosty.`repo_name.main.namespace`.table_name;
+```
+
+  </div>
+</div>
+
+### Namespaces and Tables
+
+#### Namespace Operations
+
+The Iceberg Catalog supports Iceberg namespace operations:
+
+- Create namespaces
+- List namespaces
+- Drop namespaces
+- List tables within namespaces
+
+#### Namespace Usage
+
+Namespaces in the Iceberg Catalog follow the pattern `"<repository>.<branch>.<namespace>(.<namespace>...)"` where:
+
+- `<repository>` must be a valid lakeFS repository name.
+- `<branch>` must be a valid lakeFS branch name.
+- `<namespace>` components can be nested using unit separator (e.g., `inventory.books`).
+
+Examples:
+- `my-repo.main.inventory`
+- `my-repo.feature-branch.inventory.books`
+
+The repository and branch components must already exist in lakeFS before using them in the Iceberg catalog.
+
+#### Namespace Restrictions
+
+- Repository and branch names must follow lakeFS naming conventions.
+- Namespace components cannot contain special characters except dots (.) for nesting.
+- The total namespace path length must be less than 255 characters.
+- Namespaces are case-sensitive.
+- Empty namespace components are not allowed.
+
+### Table Operations
+
+The Iceberg Catalog supports all standard Iceberg table operations:
+
+- Create tables with schemas and partitioning.
+- Update table schemas and partitioning.
+- Commit changes to tables.
+- Delete tables.
+- List tables in namespaces.
+
+### Version Control Features
+
+The Iceberg Catalog integrates with lakeFS's version control system, treating each table change as a commit. 
+This provides a complete history of table modifications and enables branching and merging workflows.
+
+#### Catalog Changes as Commits
+
+Each modification to a table (schema changes, data updates, etc.) creates a new commit in lakeFS. 
+Creating or deleting a namespace or a table results in a lakeFS commit on the relevant branch, as well as table data updates ("Iceberg table commit").
+
+#### Branching
+
+Create a new branch to work on table changes:
+
+```python
+# Create a lakeFS branch using lakeFS Python SDK
+branch = lakefs.repository('repo').branch('new_branch').create(source_reference='main')
+
+# The table is now accessible in the new branch
+new_table = catalog.load_table(f'repo.{branch.id}.inventory.books')
+```
+
+#### Merging
```

Comment:
consider merging this section with the previous so the code example will make sense more. 
the code example needs some comments to understand the current state as the sample code doesn't help much without the context.

Reply:
Did it